plugin.js

```diff
@@ -5,27 +5,80 @@ var fs = require("fs");
-    filters: [{name: 'Network Meter Plugins', extensions: ['nmp']}]}, function(data, err){
-        var fileName = data.toString();
-        fs.access(fileName, fs.R_OK, function(err) {
-            if (err) {
-                dialog.showErrorBox("File Error", "File you specified could not be accessed.");
-                return console.log(err);
+exports.bindDeleteButtons = function(state){
+    localState = state;
+    var array = document.getElementsByClassName("uninstall-button");
+    for (var i=0; i < array.length; i++) {
+        array[i].addEventListener('click', function(object) {
+            state.install.set("selected", object.currentTarget.id);
+            deleteSelectedPlugin(state);
+        });
+    }
+}
+
+exports.setupInstall = function() {
+    document.getElementById("plugin_select").onclick = function(){
+        dialog.showOpenDialog({ properties: [ 'openFile'],
+        filters: [{name: 'Network Meter Plugins', extensions: ['nmp']}]}, function(data, errors){
+            if (data != undefined) {
+                var fileName = data.toString();
+                fs.access(fileName, fs.R_OK, function(err) {
+                    if (err) {
+                        dialog.showErrorBox("File Error", "File you specified could not be accessed.");
+                        return console.log(err);
+                    }
+                });
+                installPlugin(fileName);
             }
         });
-        installPlugin(fileName);
-    });
-};
+    };
+}
 
+// TODO implement callback and error handling
 var installPlugin = function(pluginPath){
     var filename = path.basename(pluginPath, '.nmp');
     var installFolder = path.resolve('plugins') + path.sep + filename + path.sep;
 
     var zip = new AdmZip(pluginPath);
     zip.extractAllTo(installFolder);
+    pluginParser.loadPlugins(function(data){
+        // actions can only be bound to grid after full initialization
+        localState.install.set("plugins", data);
+        localState.install.update()
+    });
 };
+
+var deleteSelectedPlugin = function(state) {
+    var buttons = ["Delete", "Cancel"];
+
+    dialog.showMessageBox( {"type": "warning",
+    "title": "Delete Confirmation",
+    "message": "Are you sure you want to delete?",
+    "buttons": buttons}, function(result) {
+
+    if (buttons[result] == "Delete") {
+        var selectedInt = state.install.get("selected");
+        var name = state.install.get("plugins[" + selectedInt + "].name");
+        deletePlugin(name, function() {
+
+        //TODO find more elegant way
+        pluginParser.loadPlugins(function(data){
+            // actions can only be bound to grid after full initialization
+            state.install.set("plugins", data);
+            state.install.update()
+        });
+        });
+    }
+    });
+}
+
+var deletePlugin = function(pluginName, callback){
+    var pluginFolder = path.resolve('plugins') + path.sep + pluginName + path.sep;
```

Comment:
There is no rush, I think it's better if we fix this now. For clarity, could you provide an example of a `pluginFolder` input that would cause bad behavior?